### ISSUE TITLE
[5.x]: Call to a member function getType() on null in Craft Commerce 5.2.2 Update

### ISSUE THREAD

**Issue by apitel** [craftcms/commerce#3754]
### Description
We are having an error in one of the queries that worked before Craft Commerce 5.2.2: Call to a member function getType() on null at /var/www/html/vendor/craftcms/commerce/src/elements/Variant.php:437)"}

Comparing the code it seems that the ownerId is not anymore checked which is causing the error: 
![Image](https://github.com/user-attachments/assets/c0da21fb-8ea3-46d9-bea3-ff500d409fa8)

It looks like a new check that was put in place, that is causing Server Errors from existing Craft Commerce queries before this change was implemented. As a test, the code was hybridized (old and new), which resolved the issue:

![Image](https://github.com/user-attachments/assets/57ab008e-4ca6-45b9-92c1-8d5e3cbc8418)

### Craft CMS version

Craft Pro 5.4.10

### Craft Commerce version

5.2.2

### PHP version

8.3.10

### Operating system and version

Linux 5.15.0-124-generic

### Database type and version

MySQL 8.0.39

### Image driver and version

GD 8.3.10

### Installed plugins and versions

-

**Comment by apitel** [comment]
Full stack trace added per request:
```
2024-11-07 13:11:49 [web.ERROR] [Error] Error: Call to a member function getType() on null in /var/www/html/vendor/craftcms/commerce/src/elements/Variant.php:437
Stack trace:

#0 /var/www/html/vendor/craftcms/cms/src/base/Element.php(5152): craft\commerce\elements\Variant->getFieldLayout()

#1 /var/www/html/vendor/craftcms/cms/src/base/Element.php(5042): craft\base\Element->providerHandle()

#2 /var/www/html/vendor/craftcms/cms/src/base/Element.php(2325): craft\base\Element->hasEagerLoadedElements()

#3 /var/www/html/vendor/craftcms/cms/src/helpers/Template.php(137): craft\base\Element->__get()

#4 /var/www/html/storage/runtime/compiled_templates/76/7620bd6870546be28530e4ce633a7339.php(1138): craft\helpers\Template::attribute()

#5 /var/www/html/vendor/twig/twig/src/Template.php(393): __TwigTemplate_e9bead44d28b4fc9e62101ae000a6452->doDisplay()

#6 /var/www/html/vendor/twig/twig/src/Template.php(349): Twig\Template->yield()

#7 /var/www/html/vendor/twig/twig/src/Template.php(364): Twig\Template->display()

#8 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(35): Twig\Template->render()

#9 /var/www/html/vendor/twig/twig/src/Environment.php(306): Twig\TemplateWrapper->render()

#10 /var/www/html/vendor/craftcms/cms/src/web/View.php(500): Twig\Environment->render()

#11 /var/www/html/vendor/putyourlightson/craft-sprig-core/src/controllers/ComponentsController.php(70): craft\web\View->renderTemplate()

#12 [internal function]: putyourlightson\sprig\controllers\ComponentsController->actionRender()

#13 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()

#14 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()

#15 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()

#16 /var/www/html/vendor/craftcms/cms/src/web/Application.php(350): yii\base\Module->runAction()

#17 /var/www/html/vendor/craftcms/cms/src/web/Application.php(649): craft\web\Application->runAction()

#18 /var/www/html/vendor/craftcms/cms/src/web/Application.php(312): craft\web\Application->_processActionRequest()

#19 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()

#20 /var/www/html/public/index.php(12): yii\base\Application->run()

#21 {main} {"memory":36893680,"exception":"[object] (Error(code: 0): Call to a member function getType() on null at /var/www/html/vendor/craftcms/commerce/src/elements/Variant.php:437)"} 
```

**Comment by nfourtythree** [comment]
Hi @apitel 

Thank you for bringing this to our attention.

We have published Commerce `5.2.2.1` with a fix for this included.

Thanks!
